eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/ImmutableBag.java

```diff
@@ -120,6 +120,15 @@
     @Override
     <V> ImmutableBag<V> flatCollect(Function<? super T, ? extends Iterable<V>> function);
 
+    /**
+     * @since 8.2
+     */
+    @Override
+    default <V> ImmutableBag<V> countBy(Function<? super T, ? extends V> function)
+    {
+        return this.asLazy().<V>collect(function).toBag().toImmutable();
```

Comment:
On bags, I'd expect lazy collect to be slower than eager collect, since we haven't done anything to optimize CollectIterable for bag targets. The test should just be on a bag with a few keys where the counts are MAX_INT. Call this method, it should be slow. Then replace it with `return this.collect(function)` and it should be fast.